mrjob/spark/mrjob_spark_harness.py

```diff
@@ -72,17 +72,43 @@ def _run_step(step, step_num, rdd, make_job):
             lambda pairs: mapper_job.map_pairs(pairs, step_num))
         rdd = rdd.map(lambda k_v: m_write(*k_v))
 
+    if step_desc.get('combiner'):
+        combiner_job = make_job('--combiner', '--step-num=%d' % step_num)
+        c_read, c_write = combiner_job.pick_protocols(step_num, 'combiner')
+
+        # run the combiner
+        rdd = rdd.map(c_read)
+
+        def combiner_helper(pairs1, pairs2):
+            if len(pairs1) == len(pairs2) == 1:
+                return list(
+                    combiner_job.combine_pairs(pairs1 + pairs2, step_num),
+                )
+            else:
```

Comment:
I don't see a test that tests what happens when combiners yield zero values, or two or more. Both of these would be good candidates for a test case that simply compares output with running the job the normal way.

However, you should *also* have a separate test case that ensures that if a combiner spits out two or more values, we don't run either of those values back through the combiner; we don't want to be accidentally running poorly designed combiners in O(n^2) time.